src/main/java/io/cdap/plugin/marketo/source/batch/MarketoReportingSourceConfig.java

```diff
@@ -0,0 +1,187 @@
+import java.time.format.DateTimeParseException;
+
+/**
+ * Provides all required configuration for reading Marketo entities.
+ */
+public class MarketoReportingSourceConfig extends ReferencePluginConfig {
+  public static final String PROPERTY_CLIENT_ID = "clientId";
+  public static final String PROPERTY_CLIENT_SECRET = "clientSecret";
+  public static final String PROPERTY_REST_API_ENDPOINT = "restApiEndpoint";
+  public static final String PROPERTY_REPORT_TYPE = "reportType";
+  public static final String PROPERTY_START_DATE = "startDate";
+  public static final String PROPERTY_END_DATE = "endDate";
+
+  @Name(PROPERTY_CLIENT_ID)
+  @Description("Marketo Client ID.")
+  @Macro
+  protected String clientId;
+
+  @Name(PROPERTY_CLIENT_SECRET)
+  @Description("Marketo Client secret.")
+  @Macro
+  protected String clientSecret;
+
+  @Name(PROPERTY_REST_API_ENDPOINT)
+  @Description("REST API endpoint URL.")
+  @Macro
+  protected String restApiEndpoint;
+
+  @Name(PROPERTY_REPORT_TYPE)
+  @Description("Report type format, leads or activities.")
+  @Macro
+  protected String reportType;
+
+  @Name(PROPERTY_START_DATE)
+  @Description("Start date for the report.")
+  @Macro
+  protected String startDate;
+
+  @Name(PROPERTY_END_DATE)
+  @Description("End date for the report.")
+  @Macro
+  protected String endDate;
+
+
+  private transient Schema schema = null;
+  private transient Marketo marketo = null;
+
+  public MarketoReportingSourceConfig(String referenceName) {
+    super(referenceName);
+  }
+
+  public Schema getSchema() {
+    if (schema == null) {
+      schema = MarketoReportingSchemaHelper.getSchema(this);
+    }
+    return schema;
+  }
+
+  public Marketo getMarketo() {
+    if (marketo == null) {
+      marketo = new Marketo(getRestApiEndpoint(), getClientId(), getClientSecret());
+    }
+    return marketo;
+  }
+
+
+  public String getClientId() {
+    return clientId;
+  }
+
+  public String getClientSecret() {
+    return clientSecret;
+  }
+
+  public String getRestApiEndpoint() {
+    return restApiEndpoint;
+  }
+
+  public String getStartDate() {
+    return startDate;
+  }
+
+  public String getEndDate() {
+    return endDate;
+  }
+
+  public ReportType getReportType() {
+    return ReportType.fromString(reportType);
+  }
+
+  void validate(FailureCollector failureCollector) {
+    validateDate(failureCollector);
+    validateReportType(failureCollector);
+    validateMarketoEndpoint(failureCollector);
+    validateSecrets(failureCollector);
+  }
+
+  void validateDate(FailureCollector failureCollector) {
+    if (!(containsMacro(PROPERTY_START_DATE) && containsMacro(PROPERTY_END_DATE))) {
+      try {
+        Helpers.getDateRanges(getStartDate(), getEndDate());
+      } catch (IllegalArgumentException ex) {
+        String message = String.format("Failed to validate dates: %s.", ex.getMessage());
+        String correctiveAction = null;
+        if (ex.getMessage().contains("start date more than end date")) {
```

Comment:
what if the exception message changes? This is not the best way to do this. Maybe create our own exception or something similar.